pandas/tests/groupby/aggregate/test_aggregate.py

```diff
@@ -1377,6 +1377,43 @@ def func(ser):
     tm.assert_frame_equal(res, expected)
 
 
+@pytest.mark.parametrize(
+    "dtype",
+    [
+        "int64",
+        "uint64",
+        "int32",
+        "uint32",
+        "float64",
+        "float32",
+        "Int64",
+        "UInt64",
+        "Int32",
+        "UInt32",
+        "Float64",
+        "Float32",
+    ],
+)
+def test_groupby_agg_precision(dtype):
+    # GH33234
+    df = DataFrame(
+        {
+            "key1": ["a"],
+            "key2": ["b"],
+            "key3": [
+                pd.array([1583715738627261039], dtype=dtype)[0]
```

Comment:
Could you specify a `max_value` variable above this `df` assignment and then write this as

```
df = pd.DataFrame({..., "key3": pd.array([max_value], dtype=dtype)})
```